app/uk/gov/hmrc/helptosavereminder/util/DateTimeFunctions.scala

```diff
@@ -16,75 +16,28 @@
 
 package uk.gov.hmrc.helptosavereminder.util
 
-import java.time.{Duration, LocalDate, LocalDateTime, YearMonth, ZoneId}
-import java.util.Calendar
+import java.time.LocalDate
 
 object DateTimeFunctions {
 
-  def getNextSendDate(daysToReceive: Seq[Int]): LocalDate = {
+  def getNextSendDate(daysToReceive: Seq[Int], localDateParam: LocalDate): LocalDate = {
```

Comment:
This function is a lot better, but it still has several quirks and flaws.

I believe that it would now succeed for the values you expect it to receive, that is ones where daysToReceive is one of (1), (25) or (1, 25) and as long as the function runs on a day that you expect it to run.  Your tests now show that.

But there are two problems, on theoretical, one practical.

The theoretical problem is that here in .util, this function ought to work for any values, or at least fail in an explicit way.  Without that it is difficult to read the function and just accept that it will work for the cases that matter at the moment, it would be a lot easier if we could read it worked for all.  In one place the code sorts the days, in case they are not in order, but in another not does not.  If a days value is less than one or greater than the number of days in this month or the next then it can again fail with a bad match.

The practical problem is that if the code ever failed to run on the day you expected (either 1st or 25th), say because of an outage, then it could run on a day when it would not work.  If it was ever delayed to run on say the 30th, then other things could go wrong, it would step forward a month and still find no days.  This is the kind of unpredictable behaviour that we need to program to avoid if at all possible.

I have written a version of getNextSendDate which fixes these problems and pushed it to branch dave/DLS-3069/fix-getNextSendDate-bugs.  

But I had a better idea, on this branch dave/DLS-3069/simpler-getNextSendDate is a function that is a lot simpler, it should be immediately evident from this one that it just works, because there are no edge cases.  It is though so simple minded that it generates 62 date times.  So finally, I wrote a tailrace version which just loops until it finds a solution, on branch dave/DLS-3069/tailrec-getNextSendDate .

I would recommend that you use the code from one of those two later branches, please